frontend/modules/tree/tree.sagas.ts

```diff
@@ -62,48 +218,336 @@ export function* getSelectedNodes() {
 	}
 }
 
-export function* stopListenOnSelections() {
+function* clearCurrentlySelected() {
+	Viewer.clearHighlights();
+	yield all([
+		put(TreeActions.setActiveNode(null)),
+		call(TreeProcessing.clearSelected),
+		put(ViewerActions.setMetadataVisibility(false)),
+		put(BimActions.setActiveMeta(null))
+	]);
+	yield put(TreeActions.updateDataRevision());
+}
+
+/**
+ * SHOW NODES
+ */
+function* showAllNodes() {
 	try {
-		Viewer.off(VIEWER_EVENTS.OBJECT_SELECTED);
-		Viewer.off(VIEWER_EVENTS.MULTI_OBJECTS_SELECTED);
-		Viewer.off(VIEWER_EVENTS.BACKGROUND_SELECTED);
+		const nodesIds = yield select(selectInvisibleNodesIds);
```

Comment:
It can't be named this way. All selector's name should start with `select` - it's a common approach. If you still want to do that this way, we'll also need to change the names of the rest of selectors in the project. ;)

Reply:
the comment here is more about it should be `hidden` instead of `invisible` :P

Reply:
Ohh :D So it's fixed :P